devito/stencilkernel.py

```diff
@@ -97,7 +98,8 @@ def __init__(self, stencils, **kwargs):
 
         # Introduce C-level profiling infrastructure
         self.sections = OrderedDict()
-        nodes = self._profile_sections(nodes)
+        if self.profiler:
```

Comment:
I'll do this then..

Btw, curiosity: what's the reason you explocitly don't want C-level profiling?